contracts/ColonyTask.sol

```diff
@@ -309,13 +309,15 @@ contract ColonyTask is ColonyStorage, DSMath {
   }
 
   function finalizeTask(uint256 _id) public
-  auth
   taskWorkRatingsAssigned(_id)
   taskNotFinalized(_id)
   {
     Task storage task = tasks[_id];
     IColonyNetwork colonyNetworkContract = IColonyNetwork(colonyNetworkAddress);
 
+    // Stack too deep, so we have to reuse task variable
+    require(task.roles[MANAGER].user == msg.sender);
```

Comment:
Stack depth is becoming a real nuisance. I have some hope it maybe solved by the refactoring of this function that Daniel is doing but meanwhile can we keep the modifier and start removing the local variables, like `colonyNetworkContract` and `task`

Reply:
Daniel's PR was merged so you can test stack depth again here.

Reply:
The problem remains, and removing local variables won't help. The only way to solve this is with internal functions.